sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -271,6 +272,12 @@ def __init__(self, transform_evaluator_registry, evaluation_context,
     self._transform_evaluator_registry = transform_evaluator_registry
     self._evaluation_context = evaluation_context
     self._input_bundle = input_bundle
+    # For non-empty bundles, store the window of the max EOW
```

Comment:
<!--new_thread; commit:0d82258b7674656d3a6f86748f47d23de8cf512f; resolved:0-->
Period at end of sentence.

Can you add a TODO to modify the Bundle class to retrieve this more efficiently?

Reply:
Done